### PR TITLE
Fix: support CAL test pub key

### PR DESCRIPTION
**Pull request by sarneijim**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - ...

### 📝 Description

Enable to use test and prod provider pub key for sell

### ❓ Context

- **JIRA or GitHub link**:  https://ledgerhq.atlassian.net/browse/LIVE-14879

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: libs/ledger-live-common/src/exchange/providers/sell.ts [hunk 1]
```diff
@@ -18,44 +18,36 @@ const testSellProvider: ExchangeProviderNameAndSignature = {
   version: 2,
 };
 
-let providerDataCache: Record<string, ExchangeProviderNameAndSignature> | null = null;
-
 /**
  * The result is cached after the first successful fetch to avoid redundant network calls.
```
Comment (CremaFR) [inline libs/ledger-live-common/src/exchange/providers/sell.ts:22]: We can remove this comment, since you remove the cache
```suggestion
```

**Review by CremaFR — APPROVED** [review]

File: libs/ledger-live-common/src/exchange/providers/sell.ts (hunk 1 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/exchange/providers/sell.ts]: I have remove the complete comment. Thanks!

**Review by KVNLS — DISMISSED** [review]
Can you update file ownership, please? I think that Live Hub team might not be a reviewer for this change.

File: libs/ledger-live-common/src/cal/partners.test.ts [hunk 2]
```diff
@@ -58,10 +61,12 @@ describe("transformData", () => {
     ],
   ])(
     "should transform providers data correctly with %p env",
-    (env: string, expected: Record<string, ExchangeProvider>) => {
+    (_env: string, testConfig: boolean, expected: Record<string, ExchangeProvider>) => {
+      setEnv("MOCK_EXCHANGE_TEST_CONFIG", testConfig);
```
Comment (sprohaszka-ledger) [inline libs/ledger-live-common/src/cal/partners.test.ts:65]: Don't add env variable setting in this test.
The method you test shouldn't rely on env variable.

the `testConfig` is useless as it is redundant with the `env` parameter.

File: libs/ledger-live-common/src/cal/partners.test.ts [hunk 3]
```diff
@@ -88,7 +94,7 @@ describe("transformData", () => {
         },
       ];
 
-      const result = transformData(providersData, env as "prod" | "test");
+      const result = transformData(providersData);
```
Comment (sprohaszka-ledger) [inline libs/ledger-live-common/src/cal/partners.test.ts:97]: We need to tell the function what environment we want to use.

File: libs/ledger-live-common/src/cal/partners.test.ts [hunk 4]
```diff
@@ -98,7 +104,7 @@ describe("transformData", () => {
 
     const expected: Record<string, ExchangeProvider> = {};
 
-    const result = transformData(providersData, "prod");
+    const result = transformData(providersData);
```
Comment (sprohaszka-ledger) [inline libs/ledger-live-common/src/cal/partners.test.ts:107]: Idem

File: libs/ledger-live-common/src/cal/partners.ts [hunk 5]
```diff
@@ -49,11 +50,11 @@ export type ProvidersDataResponse = {
 // Exported for test purpose only
 export function transformData(
   providersData: ProvidersDataResponse,
-  env: "prod" | "test",
 ): Record<string, ExchangeProvider> {
+  const env = getEnv("MOCK_EXCHANGE_TEST_CONFIG") ? "test" : "prod";
```
Comment (sprohaszka-ledger) [inline libs/ledger-live-common/src/cal/partners.ts:54]: I disagree the usage of the `getEnv` function in this file. The `env` parameter` was meant to pass what value is needed, based on how we call those functions.

You should add this `getEnv` call in the `providers/swap::getSwapProvider` and `providers/index::getProviderConfig` functions. Frankly I don't recall why I didn't do it.

**Review by sprohaszka-ledger — CHANGES_REQUESTED** [review]
I disagree the usage of `getEnv` function in the cal module.
The best is to always call this method as high as possible in the call stack. The `env` parameter in the `transformData` and `getProvidersData` functions.

File: libs/ledger-live-common/src/cal/partners.test.ts (hunk 3 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/cal/partners.test.ts:97]: Not anymore, the environment variable is called directly in the function.

File: libs/ledger-live-common/src/cal/partners.test.ts (hunk 4 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/cal/partners.test.ts:107]: Idem

File: libs/ledger-live-common/src/cal/partners.ts (hunk 5 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/cal/partners.ts:54]: I do not think bring any value pass a env variable across multiple functions...

File: libs/ledger-live-common/src/cal/partners.test.ts (hunk 2 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/cal/partners.test.ts:65]: env do not have sense anymore, only set the title of the test.

File: CODEOWNERS [hunk 6]
```diff
@@ -71,6 +71,9 @@ libs/ledger-live-common/src/exchange/                                @ledgerhq/p
 libs/exchange-module/                                                @ledgerhq/ptx
 libs/wallet-api-acre-module/                                         @ledgerhq/ptx
 libs/ledgerjs/packages/hw-app-exchange/                              @ledgerhq/ptx
+libs/ledger-live-common/src/cal/partners*.ts                         @ledgerhq/ptx                    @ledgerhq/ptx
+libs/ledger-live-common/src/exchange/providers/                      @ledgerhq/ptx
```
[the remark this answers is not in the document]
Comment (sarneijim) [reply CODEOWNERS]: You are right, this git owner is redundant. Remove it

File: libs/ledger-live-common/src/cal/partners.test.ts (hunk 3 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/cal/partners.test.ts]: Changed.

File: libs/ledger-live-common/src/cal/partners.ts (hunk 5 above)
Comment (sarneijim) [reply libs/ledger-live-common/src/cal/partners.ts]: Changed.

**Review by sprohaszka-ledger — APPROVED** [review]
LGTM

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/ExchangeTestPartnerMode.tsx [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import { useTranslation } from "react-i18next";
+import { SettingsSectionRow } from "../../SettingsSection";
+
+const ExchangeTestPartnerMode = () => {
+  const { t } = useTranslation();
+  const [enableExchangeTestPartnerMode, setEnableExchangeDevMode] = useState<boolean>(false);
+
+  useEffect(() => {
+    setEnableExchangeDevMode(getEnv("MOCK_EXCHANGE_TEST_PARTNER"));
+  }, []);
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/ExchangeTestPartnerMode.tsx:14]:
```suggestion
  const [enableExchangeTestPartnerMode, setEnableExchangeDevMode] = useState<boolean>(getEnv("MOCK_EXCHANGE_TEST_PARTNER"));
```
I think that you do not need this useEffect, am I wrong?

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/ExchangeTestPartnerMode.tsx [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+  useEffect(() => {
+    setEnv("MOCK_EXCHANGE_TEST_PARTNER", enableExchangeTestPartnerMode);
+  }, [enableExchangeTestPartnerMode]);
+
+  const handleChangeSwitch = () => {
+    setEnableExchangeDevMode(!enableExchangeTestPartnerMode);
+  };
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/ExchangeTestPartnerMode.tsx:22]:
```suggestion
  const handleChangeSwitch = () => {
    setEnableExchangeDevMode(!enableExchangeTestPartnerMode);
        setEnv("MOCK_EXCHANGE_TEST_PARTNER", !enableExchangeTestPartnerMode);
  };
```

Same here, you might not need an useEffect.

**Review by KVNLS — CHANGES_REQUESTED** [review]

**Review by CremaFR — DISMISSED** [review]

**Review by KVNLS — DISMISSED** [review]

**Review by CremaFR — APPROVED** [review]

**Review by KVNLS — APPROVED** [review]

**Review by sprohaszka-ledger — APPROVED** [review]

**Pull request merged**